src/main.rs

```diff
@@ -4,31 +4,19 @@
 extern crate rocket;
 #[macro_use] extern crate rocket_contrib;
 #[macro_use] extern crate serde_derive;
-
 #[macro_use] extern crate diesel;
-extern crate r2d2;
-extern crate r2d2_diesel;
 
 use rocket_contrib::{Json, Value};
 
-mod hero;
-use hero::{Hero};
-
 mod db;
 mod schema;
 
+mod hero;
+use hero::Hero;
+
 #[post("/", data = "<hero>")]
 fn create(hero: Json<Hero>, connection: db::Connection) -> Json<Hero> {
-    let hero = hero.into_inner();
-
-    let insert = Hero {
-        id: None,
-        name: hero.name,
-        identity: hero.identity,
-        hometown: hero.hometown,
-        age: hero.age
-    };
-
+    let insert = Hero { id: None, ..hero.into_inner() };
```

Comment:
😱 I had no idea this was possible!!

Reply:
Cool, glad that both of us learn something. :laughing: 

I tried working on it myself ground up but probably couldn't do it without referencing yours. :cry: 

As well, you could do:
```rust
struct Hero {
    id: String,
    name: String,
}

let hero = Hero { id, name };

// which is equivalent to
let hero = Hero {
    id: id,
    name: name,
};
```